Studies/source/PuritySignalStudy.cxx

```diff
@@ -221,26 +208,29 @@ class SignalPurityStudy {
 
     std::vector<size_t> orderTaus(const Event& event, TauIdDiscriminator tau_id_discriminator, bool is_descending_order = true)
     {
-       std::vector<float> raw_values;
-       for(size_t lep_index = 0; lep_index < event.lep_p4.size(); ++lep_index){
+        std::vector<float> raw_values;
+        for(size_t lep_index = 0; lep_index < event.lep_p4.size(); ++lep_index){
            if (static_cast<LegType>(event.lep_type.at(lep_index)) == analysis::LegType::tau ) {
                auto lepton = std::make_shared<ntuple::TupleLepton>(event, lep_index);
                auto raw = static_cast<float>(lepton->GetRawValue(tau_id_discriminator));
-               raw_values.push_back(raw);
+               if((args.channel() == Channel::ETau || args.channel() == Channel::MuTau) && raw > lep_wp_deep_tau[args.channel()])
```

Comment:
the cut should be applied on VSe and VSmu, why do you apply cut on a raw VSjet value?